src/main/java/com/denizenscript/depenizen/bukkit/commands/quests/QuestsCommand.java

```diff
@@ -0,0 +1,221 @@
+    // @Short Edits quest player information.
+    //
+    // @Description
+    // This command allows you to give, quit, or set the stage of a quest for a player, or to add, subtract, or set
+    // the amount of Quest Points that a player holds.
+    //
+    // @Tags
+    // None
+    //
+    // @Usage
+    // Use to force player to take MyFirstQuest quest, ignoring requirements.
+    // - quests add quest-id:custom1 state:true
+    //
+    // @Usage
+    // Use to force player to quit MyFirstQuest quest, notifying player.
+    // - quests remove quest-id:custom2 state:true
+    //
+    // @Usage
+    // Use to force player into specified stage 2 of MyFirstQuest quest.
+    // - quests set quest-id:custom3 stage-no:2
+    //
+    // @Usage
+    // Use to give player 100 Quest Points.
+    // - quests add points:100
+    //
+    // -->
+
+    private enum Action {ADD, REMOVE, SET}
+    
+    private enum State {TRUE, FALSE}
+
+    public QuestsCommand() {
+
+    }
+
+    @Override
+    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
+
+        // Iterate through arguments
+        for (Argument arg : scriptEntry.getProcessedArgs()) {
+
+            if (!scriptEntry.hasObject("action")
+                    && arg.matchesEnum(Action.values())) {
+                scriptEntry.addObject("action", arg.asElement());
+            }
+
+            else if (!scriptEntry.hasObject("quest-id")
+                    && arg.matchesPrefix("quest", "quest-id")) {
+                scriptEntry.addObject("quest-id", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("stage-no")
+                    && arg.matchesPrefix("stage", "stage-no")
+                    && arg.matchesPrimitive(ArgumentHelper.PrimitiveType.Integer)) {
+                scriptEntry.addObject("stage-no", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("points")
+                    && arg.matchesPrefix("points")
+                    && arg.matchesPrimitive(ArgumentHelper.PrimitiveType.Integer)) {
+                scriptEntry.addObject("points", arg.asElement());
+            }
+            
+            else if (!scriptEntry.hasObject("state")
+                    && arg.matchesPrefix("state")
+                    && arg.matchesEnum(State.values())) {
+                scriptEntry.addObject("state", arg.asElement());
+            }
+            else {
+                arg.reportUnhandled();
+            }
+
+        }
+
+        if (!scriptEntry.hasObject("action")) {
+            throw new InvalidArgumentsException("Must specify a valid action!");
+        }
+        
+        scriptEntry.defaultObject("state", new ElementTag("TRUE"));
+    }
+
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        Quests quests = (Quests) QuestsBridge.instance.plugin;
+        
+        BukkitScriptEntryData scriptEntryData = (BukkitScriptEntryData) scriptEntry.entryData;

```

Comment:
This variable is redundant, as it's only used once.

Reply:
Once per command, yes, but referenced in nine places throughout the method. How would you prefer this be handled?

Reply:
It's referenced in exactly one place - line 127.

Reply:
Oops, thought you meant the Quests variable, but I see you mean BukkitScriptEntryData. Will address.